### ISSUE TITLE
Allow some third-party integrations 

### ISSUE THREAD

**Issue by sebastian-lung** [GoogleChromeLabs/privacy-sandbox-dev-support#90]
Is there planned any feature that would allow some third-party integrations that are common on the web to continue to function?

For example, I see on Firefox a "transitional feature": Storage Access Heuristics 
https://developer.mozilla.org/en-US/docs/Web/Privacy/State_Partitioning#storage_access_heuristics

I am responsible of a web authentication system integrated by plenty of external partners on different domains than ours using iframes or popups integrations. On top of that, most of these partners are business strategic assets.

This Storage Partitioning feature would break the authentication and Single-Sign-On on all our cross-domain integrations.

**Comment by sebastian-lung** [comment]
Hello @rowan-m, thank you for your answer!

We are using only localStorage with third parties as functionality that would break.

The suggested approach https://developer.chrome.com/en/docs/privacy-sandbox/fedcm/ is interesting and good to have it in mind for long term evolutions. However, it is not an option with the current architecture we have.

**Comment by miketaylr** [comment]
We do plan to offer a deprecation trial (and hope to have our blog post up very soon), so that may be an option for you. If you want to test things in the latest Canary (and are happy to wade in the dark for a bit w/o the detailed instructions):

[DisableThirdPartySessionStoragePartitioningAfterGeneralPartitioning](https://developer.chrome.com/origintrials/#/view_trial/3444127815031586817)

[DisableThirdPartyStoragePartitioning](https://developer.chrome.com/origintrials/#/view_trial/-8517432795264450559)

**Comment by miketaylr** [comment]
It would also be helpful if you could highlight the general patterns of how localStorage is used in these 3P integrations, thanks.

**Comment by miketaylr** [comment]
Also, do these sites happened to be used in enterprise/managed environment contexts, @sebastian-lung ?

**Comment by sebastian-lung** [comment]
Hello @miketaylr,

Thank you very much for the details provided.

Regarding the deprecation trial, as far as I understand in order to work each domain would need to make a request to opt into unpartitioned storage. Considering we have a very big and dynamic list of integrators this would be problematic.

Please find bellow general patterns of how localStorage is used in these 3P integrations:
Our web authentication system is represented by a webpage that is hosted on the main domain, sub.x.com and a JS SDK that partners integrate to authenticate the users.
This JS SDK is integrated on multiple domains that are different than x.com and it communicates with a hidden iFrame on sub.x.com to retrieve the user session.
The user session it's always stored in the localstorage of sub.x.com. Before the Storage Partitioning feature, there were no issues when an integrator on domain y.com used the JS SDK to retrieve the user session from the localstorage of sub.x.com but after the update this won't be possible anymore.
So to summarize, in our context, the localstorage is used by sub.x.com to store the user session after it's obtained, and by partners to retrieve it from sub.x.com through the JS SDK.

And for the last questions, our system is for a private company, though it is hosted online on the web and used by end-users (any user on the web). Integration sites are own either by our company or by business partners (usually other private companies).

**Comment by miketaylr** [comment]
This reminds me of https://github.com/privacycg/storage-partitioning/issues/34#issuecomment-1194358637 - but Firebase is/was using sessionStorage instead of localStorage.

**Comment by miketaylr** [comment]
@sebastian-lung how does this work in Safari (does it?)?

**Comment by sebastian-lung** [comment]
The mechanism described previously does not work in Safari, and also not on latest Firefox versions with partition storage on cross-domain. We need workarounds for these cases.
Chrome is used by more than 80% of our users so this is the most important browser for us.

**Comment by miketaylr** [comment]
@sebastian-lung could you check out https://developer.chrome.com/blog/storage-partitioning-deprecation-trial/ and see if that solves your use cases?

**Comment by sebastian-lung** [comment]
I think it is the feature you mentioned before, and I said it would be problematic if we need to make an opt-in request for each domain as we have a big dynamic list.

**Comment by miketaylr** [comment]
How big is the list? Can you share an approximate answer?

**Comment by miketaylr** [comment]
It may be useful to jump on a video call to discuss your use case a bit more, feel free to email miketaylr@google.com and we can set something up.

**Comment by sebastian-lung** [comment]
> How big is the list? Can you share an approximate answer?

The list contains about 300 entries.

**Comment by miketaylr** [comment]
Hi @sebastian-lung, FYI we sent our Intent to Ship this week - the details of our intended rollout are listed as well:

https://groups.google.com/a/chromium.org/g/blink-dev/c/24hK6DKJnqY/m/ChL2WWx5CgAJ

Feel free to reach out to my email listed above and we can chat about the impact to your use case.

thanks,
Mike

**Comment by miketaylr** [comment]
@sebastian-lung FYI, we added support for a 3rd party deprecation trial (see https://developer.chrome.com/blog/storage-partitioning-deprecation-trial/#participate-in-the-deprecation-trials), and plan to start ramping this up in Chrome 115. 

Hope that helps.
